src/client/connect.rs

```diff
@@ -246,6 +254,16 @@ mod http {
         pub fn set_nodelay(&mut self, nodelay: bool) {
             self.nodelay = nodelay;
         }
+
+        /// Set that all sockets are bound to the configured address before connection.
+        ///
+        /// If `None`, the sockets will not be bound.
+        ///
+        /// Default is `None`.
+        #[inline]
+        pub fn set_local_address(&mut self, addr: Option<IpAddr>) {
```

Comment:
Just some bikeshedding: what is something like this more commonly called? A "local" address? The "bind" address? And, would it be better to call this an "address", or just "ip"?

I don't have much preference myself, as long as it's been considered.

Reply:
It's normally called the local address; see bind(2) at https://linux.die.net/man/2/bind and http://pubs.opengroup.org/onlinepubs/009695399/functions/bind.html . Normally `bind` takes a `SockAddr`, but for this specific case of binding-before-connecting, it's vanishingly rare that you want to specify the port as well as the IP, which is why I have specified `IpAddr`. But it's still an address.